File: src/main/java/io/github/oleksivio/telegram/bot/api/annotations/filter/telegram/GameFilter.java
package io.github.oleksivio.telegram.bot.api.annotations.filter.telegram;

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.StringFilter;
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
@Documented
public @interface GameFilter {

    AnnotationState status() default AnnotationState.ON;

    StringFilter title() default @StringFilter(status = AnnotationState.OFF);

    StringFilter description() default @StringFilter(status = AnnotationState.OFF);

    PhotoArrayFilter photoArray() default @PhotoArrayFilter(status = AnnotationState.OFF);

    StringFilter text() default @StringFilter(status = AnnotationState.OFF);

    MessageEntityFilter textEntities() default @MessageEntityFilter(status = AnnotationState.OFF);

    AnimationFilter animation() default @AnimationFilter(status = AnnotationState.OFF);
}

File: src/main/kotlin/io/github/oleksivio/telegram/bot/api/annotations/filter/composite/GameFilter.kt
package io.github.oleksivio.telegram.bot.api.annotations.filter.composite

import io.github.oleksivio.telegram.bot.api.annotations.filter.primitive.StringFilter
import io.github.oleksivio.telegram.bot.api.model.annotation.AnnotationState

@Retention
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@MustBeDocumented
annotation class GameFilter(val status: AnnotationState = AnnotationState.ON, val validator: Array<String> = [], val title: StringFilter = StringFilter(status = AnnotationState.OFF), val description: StringFilter = StringFilter(status = AnnotationState.OFF), val photoArray: PhotoArrayFilter = PhotoArrayFilter(status = AnnotationState.OFF), val text: StringFilter = StringFilter(status = AnnotationState.OFF), val textEntities: MessageEntityFilter = MessageEntityFilter(status = AnnotationState.OFF), val animation: AnimationFilter = AnimationFilter(status = AnnotationState.OFF))
